### ISSUE TITLE
GPL-licensed dependency `bashlex`

### ISSUE THREAD

**Issue by hehao98** [pypa/cibuildwheel#1484]
### Description

Hello, 

We are a group of researchers developing tools to monitor and remediate open-source license incompatibilities in the PyPI ecosystem.

We find that your `cibuildwheel` package has a **GPL-3.0** licensed dependency `bashlex`. This could be problematic because GPL-licensed software requires any of its derivative work to be also licensed under GPL, but `cibuildwheel` is licensed under BSD 2-clause.

To remove this license incompatibility, the following possible remediations may be considered:
* Relicense `cibuildwheel` under GPL 3.0 or AGPL 3.
* Remove `bashlex` (migrate to another alternative if necessary)

Note that the above remediations are generated by an automated tool that is still under test, may be incorrect, and does not represent legal advice. We welcome any suggestions and feedback!

Thank you.

**Comment by Czaki** [comment]
BSD 2-clause is GPL-3.0 compatible license. 
I think that we could close this issue as non-relevant.

**Comment by joerick** [comment]
Just noticed that this was generated by an automated tool. Feels like a time-sink that I'm not too interested in jumping into!

**Comment by hehao98** [comment]
Thank you for your feedback. 

Just for some clarification, the GPL license explicitly says that its "infectiveness" applies to programs calling APIs from GPL-licensed libraries (see this [FAQ](https://www.gnu.org/licenses/gpl-faq.en.html#GPLPlugins) and this [FAQ](https://www.gnu.org/licenses/gpl-faq.html#IfLibraryIsGPL)) but does not apply to programs that use syscalls to Linux (see this [FAQ](https://www.gnu.org/licenses/gpl-faq.en.html#CanIUseGPLToolsForNF) and this [discussion](https://news.ycombinator.com/item?id=20138971)). BSD 2-clause is compatible with GPL-3.0 *in the sense that they can be combined, but the combined work must be licensed user GPL-3.0* (see this [FAQ](https://www.gnu.org/licenses/gpl-faq.en.html#WhatDoesCompatMean)). 

In other words, if `cibuildwheel` calls APIs from `bashlex`, it means that they form combined work and the restrictions of GPL license will apply. 

For your information, many other popular Python packages have devoted efforts to avoid GPL-licensed dependencies to avoid adhering to the restrictions of GPL (e.g., see https://github.com/ansible/ansible-lint/issues/1188 https://github.com/facebook/prophet/issues/1069 https://github.com/jitsi/jiwer/issues/69 https://github.com/Unidata/netcdf4-python/issues/1073 https://github.com/chartbeat-labs/textacy/issues/62).

**Comment by Czaki** [comment]
The reason why this is not big problem for cibuildwheel is not a library but rather the workflow manager. And even if someone assumes that the real license is GPL, then it will not impact produced wheels. 

And this is not typical for many python packages. 

The second problem is that FAQ that you linked used names proper for compiled languages, not interpreted ones. But I do not see proper examples of GPL violation in interpreted languages. 

Of course if we would like to pack it using pyinstaller or a similar tool then produced executable will be GPL licensed. 

I think that if for someone this license worries are blocking and would like to fund implementation required `bashlex` utils then it may be worth sitting to this problem again. 

But based on my knowledge there is no such problem for cibuildwheel.
